src/t-i18n.ts

```diff
@@ -19,20 +19,20 @@ import parseReact from "./react";
 //
 export interface TFunc {
 	(message: string, replacements?: Replacements, id?: string): string;
-	lookup?: (id: string, replacements?: Replacements, defaultMessage?:string) => string 
-	setup?: (options?: SetupOptions) => any;
-	date?: (value: any, formatName?: string, locale?: string) => string;
-	number?: (value: any, formatName?: string, locale?: string) => string;
-	$?: (message: string, replacements?: ReactReplacements, id?: string) => React.ReactNode[];
+	lookup: (id: string, replacements?: Replacements, defaultMessage?:string) => string 
+	setup: (options?: SetupOptions) => any;
+	date: (value: any, formatName?: string, locale?: string) => string;
+	number: (value: any, formatName?: string, locale?: string) => string;
+	$: (message: string, replacements?: ReactReplacements, id?: string) => React.ReactNode[];
 	_i18nInstance?: I18n;
 }
 
 export const defaultLanguage = "en";
 
 export class I18n {
-	locale: string;	
-	messages: Messages;
-	idGenerator: (message: string) => any;
+	locale!: string;	
+	messages!: Messages;
+	idGenerator!: (message: string) => any;
```

Comment:
I really don't like this. It's like using `as any`. Can we not set default values in the constructor?